### PR TITLE
fix(sdcm/docker.py): fix docker and non-docker teardown proceedure

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/0hTlUE6b/1610-make-provision-test-work-for-docker-backend

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -1262,13 +1262,26 @@ def get_truncated_time_from_system_truncated(self, session, table_id):  # pylint
...
 
     def finalize_test(self):
-        try:
-            self.stop_resources()
-        finally:
-            self.save_email_data()
-            self.collect_logs()
-            self.clean_resources()
-            self.send_email()
+        steps = [
```
Comment (bentsi) [inline sdcm/tester.py]: convert to separate PR

File: sdcm/cluster.py [hunk 2]
```diff
@@ -4211,10 +4210,11 @@ def set_grafana_annotations(self, node, annotations_data):
         self.log.info("posting annotations result: %s", res)
 
     def stop_scylla_monitoring(self, node):
-        kill_script = dedent("""
-            cd {0.monitor_install_path}
+        kill_script = dedent(f"""
```
Comment (bentsi) [inline sdcm/cluster.py]: undo the change

File: sdcm/docker.py [hunk 3]
```diff
@@ -269,6 +261,7 @@ def node_setup(self, node, verbose=False, timeout=3600):
         seed_address = ','.join(self.seed_nodes_ips)
 
         node.wait_ssh_up(verbose=verbose)
+        node.run_startup_script()
```
Comment (bentsi) [inline sdcm/docker.py]: remove

Comment (dkropachev) [reply sdcm/docker.py]: removed

File: sdcm/cluster.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/cluster.py]: undone
